staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -50,10 +50,10 @@ import (
 	"k8s.io/klog/v2"
 )
 
-// ObjectFunc is a function to act on a given object. An error may be returned
-// if the hook cannot be completed. An ObjectFunc may transform the provided
-// object.
-type ObjectFunc func(obj runtime.Object) error
+// ObjectFunc is a function to act on a given object. Such hooks may not fail,
+// and do not have any way to indicate an error.  An ObjectFunc may transform
+// the provided object.
+type ObjectFunc func(obj runtime.Object)
```

Comment:
This change breaks OpenShift (we use errors from ObjectFuncs today).  Today these hooks can fail, why does this need to change now? Perhaps you need a different kind of hook?  Our decorators can return errors for instance: extreme edges, but they do so today.

Reply:
If they fail on a mutating operation, the object is persisted in etcd anyway, right?  That seems like a pretty awkward state to leave users.  What failures are really meaningful at this late stage of processing?

Reply:
> If they fail on a mutating operation, the object is persisted in etcd anyway, right? That seems like a pretty awkward state to leave users. What failures are really meaningful at this late stage of processing?

Decorators can be used to place dynamic information on the return value and may fail.  While the data is persisted, returning the unmutated data back could return incomplete or incorrect information.  We've cleaned out most egregious examples, but at least one still exists.

Reply:
As discussed today, we agreed to keep this change.